atuin-client/src/database.rs

```diff
@@ -179,33 +186,58 @@ impl Database for Sqlite {
     }
 
     // make a unique list, that only shows the *newest* version of things
-    async fn list(&self, max: Option<usize>, unique: bool) -> Result<Vec<History>> {
+    async fn list(
+        &self,
+        filter: FilterMode,
+        max: Option<usize>,
+        unique: bool,
+    ) -> Result<Vec<History>> {
         debug!("listing history");
+        let session = env::var("ATUIN_SESSION")
+            .expect("failed to find ATUIN_SESSION - check your shell setup");
+        let hostname = format!("{}:{}", whoami::hostname(), whoami::username());
+        let cwd = match env::current_dir() {
+            Ok(dir) => dir.display().to_string(),
+            Err(_) => String::from(""),
+        };
```

Comment:
Should this really be in the db layer

Reply:
Meh probably not, but also I don't want to get to the point where we're passing around 9999999 parameters to every database function 🙃

Reply:
I have some thoughts but I can experiment later

Reply:
It's chill, I'm wrapping it up into a struct I think. Those three values are rarely used in isolation, and I should stop being lazy